src/include/sof/drivers/idc.h

```diff
@@ -82,11 +82,21 @@
 /** \brief Decodes IDC message type. */
 #define iTS(x)	(((x) >> IDC_TYPE_SHIFT) & IDC_TYPE_MASK)
 
+/** \brief Max IDC message payload size in bytes. */
+#define IDC_MAX_PAYLOAD_SIZE	96
+
+/** \brief IDC message payload. */
+struct idc_payload {
+	uint8_t data[IDC_MAX_PAYLOAD_SIZE];
+};
+
```

Comment:
How do you ensure you never have two cores sending to the same target core at the same time, overwriting payloads?

Reply:
We don't. It something that probably should be done in the future, but right now we are working in master-slave mode, so only master core is sending messages to other cores.

Reply:
I'd say it's a high priority issue because you could potentially have weird arrangements with components, like pipeline 1 on core 0, pipeline 2 on core 1, and pipeline 1 has a component on core 1 and pipeline 2 has component on core 0. Topology is coming from userspace so we should be ready for this (even if by simply rejecting unsupported configurations initially).

Reply:
If I understand correctly @tlauda meant that messages are delivered from core 0 to other cores, not that it supports only pipelines on core 0 having components on other cores. If it's true then there is no issue atm.

Reply:
I thought from the pull request that messages will come from the actual core that owns the pipeline and not the core 0?

Reply:
@paulstelian97 Messages are always coming from master core, which is the only one receiving IPCs.